scripts/superdesk/ui/ui.js

```diff
@@ -536,16 +537,39 @@
      * <div sd-datepicker ng-model="date"></div>
      *
      */
-
-    function DatepickerDirective() {
+    DatepickerDirective.$inject = ['$document'];
+    function DatepickerDirective($document) {
         return {
             scope: {
                 dt: '=ngModel',
                 disabled: '=ngDisabled'
             },
             templateUrl: 'scripts/superdesk/ui/views/sd-datepicker.html',
-            link: function(scope, elem) {
+            link: function(scope, element) {
                 scope.state = {opened: false};
+                scope.openCalendar = function(e) {
+                    scope.state.opened = !scope.state.opened;
+
+                    $document.on('click', handleDatePicker);
+                };
+
+                function close() {
+                    scope.state.opened = false;
+                    $document.off('click', handleDatePicker);
+                }
+
+                function handleDatePicker(e) {
+                    var isChild = element.find(event.target).length > 0;
```

Comment:
just wondering - is there a need for this if you added this `stopPropagation` to the other directive?